### ISSUE TITLE
Legend is overlapping chart

### ISSUE THREAD

**Issue by TuringLovesDeathMetal** [lightdash/lightdash#3474]
### Description

The legend is overlapping my chart. 

We should shrink the chart to some minimum vertical size before we start overlapping the values. 

Is there actually an option to set a limit to some minimum vertical size, then add a scroll functionality for the legend?

<img width="607" alt="image" src="https://user-images.githubusercontent.com/31848148/195176691-4cd92870-e7f4-419c-b12d-ccb1197d358f.png">

**Comment by rephus** [comment]
It is possible to add `scroll` to the legend https://echarts.apache.org/en/option.html#legend.type, so instead  of this: 

![image](https://user-images.githubusercontent.com/1983672/200545080-51eca0b1-67f3-4335-a990-06687f182fc7.png)

it looks like this 
[Screencast from 08-11-22 11:47:38.webm](https://user-images.githubusercontent.com/1983672/200545187-e273a36d-0b44-4803-8043-ab90d9048859.webm)

**Comment by rephus** [comment]
There is already an echarts issue to automatically fix this by the way https://github.com/apache/echarts/issues/14252

**Comment by rephus** [comment]
Related old ticket https://github.com/lightdash/lightdash/issues/541

**Comment by TuringLovesDeathMetal** [comment]
> it is possible to add scroll to the legend. 

^I feel like this is a good solution for now? 

And then if you try to export the chart, then we make teh legend overlap (otherwise you won't see the entire legend). 

Then, we can open a separate issue which is "if you try to export a chart, the legend overlaps" and link the echarts issue there so we stay aware of it. 

Waht do you think @rephus ?

**Comment by rephus** [comment]
I think we can add conditional scrolling, so if we have more than ¿5? series, it will add scrolling, otherwise, we'll keep it as it is. This could still cause overlapping if some series have long names...

Should we make it configurable on the config panel ? 

I'm not sure how easy it is to just disable scrolling on export, but you're right, we need to show all series when exporting.

**Comment by TuringLovesDeathMetal** [comment]
Ya, that sounds like a good plan :) We can test out the number of appropriate series to see what looks right, and great idea for making it a config!

We can add it as a toggle in the legend config panel, here: 
<img width="350" alt="image" src="https://user-images.githubusercontent.com/31848148/201066350-7a225a9d-a7a7-4912-bbbf-def56881cb73.png">


Something like `scrolling` (yes/no)

**Comment by rephus** [comment]
Ahh, I didn't notice , but we already support `scroll` config on legend panel... 

![image](https://user-images.githubusercontent.com/1983672/201099586-e4aef2d2-5018-4135-b8eb-3f188b5074a6.png)

So maybe we don't need to do anything here

**Comment by TuringLovesDeathMetal** [comment]
lol, no way, I never even noticed that 😂 

Ya, I feel like we should set it to scroll automatically? 

Maybe also changing the name form `type` to `scroll` and making it a checkbox? (or toggle)
